crowbar_framework/app/models/node.rb

```diff
@@ -850,6 +850,27 @@ def file_exist?(file)
     out[:exit_code].zero?
   end
 
+  def upgraded?
+    ret = file_exist? "/var/lib/crowbar/upgrade/node-upgraded-ok"
+    Rails.logger.info("Node #{@node.name} was already upgraded.") if ret
+    ret
+  end
+
+  def upgrading?
+    file_exist? "/var/lib/crowbar/upgrade/node-upgrading"
+  end
+
+  def set_upgraded
+    run_ssh_cmd(
+      "touch /var/lib/crowbar/upgrade/node-upgraded-ok && " \
+      "rm -f /var/lib/crowbar/upgrade/node-upgrading"
+    )
+  end
+
+  def set_upgrading
+    run_ssh_cmd("touch /var/lib/crowbar/upgrade/node-upgrading")
```

Comment:
I'm trying to avoid `node.save`, don't you think it is slower than ssh command? (Also, I would go directly for attributes, but as we are already using files during the upgrade, this feels like consistent with it)